coffee-module/coffee-module-redisstream/src/main/java/hu/icellmobilsoft/coffee/module/redisstream/action/EventControlAction.java

```diff
@@ -0,0 +1,195 @@
+            }
+
+            // If the list has exactly one element after the rpush operation, we create or update the unique key used by the pipe consumer.
+            // This identifier ensures that the pipe consumer can decide whether to continue the processing loop or stop,
+            // as a new event is expected to arrive.
+            Pipeline pipeline = redisManager.run(Jedis::pipelined, "pipelined serialStreamEvent setex").orElseThrow();
+            pipeline.setex(createPipeIdKey(key), secondsToExpire, RandomUtil.generateToken());
+            pipeline.expire(key, secondsToExpire, ExpiryOption.NX);
+            pipeline.sync();
+        } finally {
+            redisManager.closeConnection();
+        }
+
+        // Flow extension configuration to prevent logs from being grouped under a single SID
+        Map<String, String> parameters = new HashMap<>(defaultMessageProperties());
+        parameters.put(IRedisStreamConstant.Common.DATA_MESSAGE_TYPE, messageType().name());
+        if (StringUtils.isNotBlank(flowExtensionId)) {
+            parameters.put(StreamMessageParameter.FLOW_ID_EXTENSION.getMessageKey(), flowExtensionId);
+        }
+
+        // If exactly one item has been added to the list, the event is published with the list's key
+        publish(key, parameters);
+        return true;
+    }
+
+    /**
+     * Handles FIFO logic. If the FIFO list is created with its first item, it publishes a message to the stream for processing. If the inserted item
+     * is not the first one, the message is not published.
+     *
+     * @param key
+     *            the key of the FIFO list
+     * @param eventMessage
+     *            the content of the FIFO list item, which will be converted to JSON format
+     * @param flowExtensionId
+     *            extension of the extSessionId
+     * @return {@literal true} if a publish occurred
+     * @throws BaseException
+     *             if any error occurs
+     */
+    public boolean serialStreamEvent(String key, T eventMessage, String flowExtensionId) throws BaseException {
+        return serialStreamEvent(key, eventMessage, getTtl(), flowExtensionId);
+    }
+
+    /**
+     * publish message
+     * 
+     * @param key
+     *            Message in stream. Can be String or JSON
+     * @param parameters
+     *            Message parameters, nullable. Map key value is standardized in {@link StreamMessageParameter} enum value
+     * @throws BaseException
+     *             if any error occurs
+     */
+    protected void publish(String key, Map<String, String> parameters) throws BaseException {
+        getRedisStreamPublisher().publish(key, parameters);
+    }
+
+    /**
+     * returns with TTL
+     * 
+     * @return the ttl
+     */
+    protected long getTtl() {
+        return getRedisStreamPublisher().getStreamGroupConfig().getProducerTTL().orElse(-1L);
+    }
+
+    private String createPipeIdKey(String key) {
+        return IRedisStreamConstant.Pipe.ID + key;
```

Comment:
Esetleg alkalmazhatnánk a redis által ajánlott névkonvenciót, hogy ne minden fejlesztés más formátumú kulcsokat használjon.
Itt pl: `"streampipe:" + key` és `"streampipe:" + key + ":id"`
A prefix csak egy példa, lehet más is ha találunk tömörebbet. Vagy ha redis cluster-t is akarunk támogatni:  `"streampipe:{" + key + "}:id"`